### DISCUSSION TITLE
[Feature] External library setup is completely inside the web UI

### DISCUSSION THREAD

**Discussion by Conor-Collins** [immich-app/immich#6121] [Feature Request]
### The feature

Just went through the process to setup external libraries.  

Currently its a little nebulous with pathing in 3-4 places for the same path depending on how you have things setup,  would be quite a thrill to have this cleaned up so that there is less duplication of pathing -  accounts/library/import path and user/edit/external path are seemingly the same requirement.  

If you could set up the library without editing the compose and make whole process doable from the Web UI alone would be the feature I would love to see.

I don't know if that is possible considering how docker volumes work but hey,  what if?

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
The compose file has to be edited for Immich to get access to the files, so simplifying that is not possible. The two different path settings are for security reasons - if that was not there, the user could enter any path they wanted and thus get access to other users' files. One possible solution there is to make external libraries be configured entirely by the admin: https://github.com/immich-app/immich/discussions/4500#discussioncomment-7892234
